controllers/nodehealthcheck_controller.go

```diff
@@ -97,6 +107,43 @@ func (r *NodeHealthCheckReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		}
 		return result, err
 	}
+
+	// check if we need to disable NHC because of existimg MHCs
+	disable, err := r.mhcChecker.NeedDisableNHC()
+	if err != nil {
+		log.Error(err, "failed to check for MHCs")
+		return result, err
+	}
+	if disable {
+		log.Info("disabling NHC in order to avoid conflict with custom MHCs configured in the cluster")
+		// update status
+		if !meta.IsStatusConditionTrue(nhc.Status.Conditions, remediationv1alpha1.ConditionTypeDisabled) {
+			meta.SetStatusCondition(&nhc.Status.Conditions, metav1.Condition{
+				Type:    remediationv1alpha1.ConditionTypeDisabled,
+				Status:  metav1.ConditionTrue,
+				Reason:  remediationv1alpha1.ConditionReasonDisabledMHC,
+				Message: "Custom MachineHealthCheck(s) detected, disabling NHC to avoid conflicts",
+			})
+			r.recorder.Eventf(&nhc, eventTypeWarning, eventReasonDisabled, "Custom MachineHealthCheck(s) detected, disabling NHC to avoid conflicts")
+			err = r.Client.Status().Update(context.Background(), &nhc)
+			if err != nil {
+				log.Error(err, "failed to update NHC status conditions")
+				return result, err
+			}
+		}
+	} else {
+		log.Info("re-enabling NHC, no conflicting MHC configured in the cluster")
```

Comment:
Might be a bit noisey to log every time. Maybe move into the IsStatusConditionTrue() block?

Reply:
yep 👍🏼